platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -139,28 +157,37 @@ public Response sendLogoutRequest(@QueryParam("EncryptedNameIdTime") String encr
           LOGGER.info(msg);
           return buildLogoutResponse(msg);
         }
+
+        // Logout removes the SAML assertion. This statement must be called before the SAML
+        // assertion is
+        // removed.
+        List<String> sessionIndexes =
+            getIdpSecurityAssertion()
+                .getAuthnStatements()
+                .stream()
+                .map(AuthnStatement::getSessionIndex)
+                .collect(Collectors.toList());
+
```

Comment:
New code. Collect the session indexes from the AuthnStatements (FYI, normally there is only one AuthnStatement) and pass them to the factory method that creates the LogoutRequest.